docs/developing/on-balena.markdown

```diff
@@ -0,0 +1,177 @@
+# Developing on balenaOS
+
+## ToC
+* [Dependencies](#dependencies)
+* [Provisioning balenaOS](#provisioning)
+* [Developing with Livepush](#development)
+* [Running e2e/integration tests](#testing)
+* [Footnotes](#footnotes)
+
+
+## Dependencies
+> all reference commands assume a Linux/BSD compatible execution environment and a Bash
+  shell.
+
+* Intel NUC or another amd64 device capable of running balenaOS[[fn1](#fn1)]
+* a balenaCloud account with access to [jellyfish-on-balena] fleet
+* balena CLI, openssl, jq, yq v4+ and Etcher[[fn3](#fn3)]
+* single network broadcast domain containing the development environment[[fn4](#fn4)]
+
+
+## Provisioning
+* request access to [jellyfish-on-balena] fleet and provision your device into it
+* login to balenaCloud
+
+```sh
+balena login
+```
+
+* find your device and configure environment
+
+```sh
+set -a
+
+tld=ly.fish.local
+
+mkdir -p balena
+
+uuid=$(printf "results:\n$(sudo balena scan)" \
+  | yq e '.results[] | select(.osVariant=="development").host' - \
+  | awk -F'.' '{print $1}') \
+  && balena_device_uuid=$(balena device ${uuid::7} | grep UUID | cut -c24-)
+```
```

Comment:
This block assumes `bash` and doesn't work with `zsh`. We should probably move this to a script that works with both and make it callable from `npm run ...`.

Reply:
it's really just a change to `... ${uuid:0:7}` which makes it work in both bash and zsh.